Core/Object Arts/Dolphin/IDE/Base/ClassBrowserAbstract.cls

```diff
@@ -114,7 +114,7 @@ actualClassChainMethods
 	^dict values!
 
 addCategoriesOfMethod: aCompiledMethod
-	(aCompiledMethod realCategories difference: categoriesPresenter model asSet) 
+	(aCompiledMethod categories difference: categoriesPresenter model asSet) 
```

Comment:
CompiledMethod>>categories is slow because it evaluates all the virtual categories. This method is evaluated in response to new methods being compiled into the system (or removed or categorised), and so slowing it down is going to have a big impact on operations such as performing file ins with browsers open. This is why the browsers haven't dynamically changed the set of virtual categories shown in the past. This needs to somehow skip categories which return true to `#isSlowToEnumerate`.
Either that or don't worry about the fact that the virtual categories may get stale.